saleor/plugins/email_common.py

```diff
@@ -273,6 +273,32 @@ def validate_default_email_configuration(plugin_configuration: "PluginConfigurat
         )
 
 
+def validate_format_of_provided_templates(
+    plugin_configuration: "PluginConfiguration", template_fields: List[str]
+):
+    """Make sure that the templates provided by the user have the correct structure."""
+    configuration = plugin_configuration.configuration
+    configuration = {item["name"]: item["value"] for item in configuration}
+
+    if not plugin_configuration.active:
+        return
+    compiler = pybars.Compiler()
+    errors = {}
+    for field in template_fields:
+        template_str = configuration.get(field)
+        if not template_str or template_str == DEFAULT_EMAIL_VALUE:
+            continue
+        try:
+            compiler.compile(template_str)
+        except pybars.PybarsError:
+            errors[field] = ValidationError(
+                "The provided template has an inccorect structure.",
+                code=PluginErrorCode.INVALID.value,
+            )
```

Comment:
What do you mean by "incorrect structure"? This error is very generic and I'm wondering if we can make it more precise.

Reply:
Instead of generic `INVALID` error code I would add some specific error e.g. `INVALID_EMAIL_TEMPLATE`. Since `ValidationError` can take additional context, we could pass the error from Handlebars compiler as text to our error, and later display the error in the dashboard. This would help the user to understand what is wrong with the template and fix it.